org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/NullAnnotationTests21.java

```diff
@@ -1244,4 +1238,46 @@ static Stuff match(PatternMatching pm, int v) {
 				""";
 		runner.runNegativeTest();
 	}
+
+	// https://github.com/eclipse-jdt/eclipse.jdt.core/issues/3381
+	// [Enhanced Switch][Null] Missing Null pointer access warning with total/unconditional patterns
+	public void testIssue3381() {
+		Runner runner = getDefaultRunner();
+		runner.testFiles = new String[] {
+			"X.java",
+			"""
+			import org.eclipse.jdt.annotation.Nullable;
+
+			public class X {
+				void foo() {
+					@Nullable Integer i = null;
+					switch (i) {
+						case Integer ii -> System.out.println();
+					}
+				}
+				void goo() {
+					@Nullable Integer i = null;
+					switch (i) {
+						default -> System.out.println();
+					}
+				}
+			}
+			"""
+		};
+		runner.expectedCompilerLog =
+				"""
+				----------
+				1. ERROR in X.java (at line 6)
+					switch (i) {
+					        ^
+				Null pointer access: The variable i can only be null at this location
+				----------
+				2. ERROR in X.java (at line 12)
+					switch (i) {
+					        ^
+				Null pointer access: This expression of type Integer is null but requires auto-unboxing
+				----------
```

Comment:
@srikanth-sankaran  do you have an idea why these are different error messages?

Reply:
> @srikanth-sankaran do you have an idea why these are different error messages?

Yes, I had looked into this. In the former case its is a type switch which doesn't call for unboxing and in the latter switch on the unboxed integer. This results in a special message in the latter case.